src/components/UserInfo.js

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react';
```

Comment:
UserInfo doesn't seem like an appropriate name for what this component does - maybe `DoiInfo`?

Reply:
It also seems really similar to manualAdd - can these be combined somehow?